operator/internal/manifests/storage/configure.go

```diff
@@ -43,26 +34,27 @@ func ConfigureDeployment(d *appsv1.Deployment, opts Options) error {
 
 // ConfigureStatefulSet appends additional pod volumes and container env vars, args, volume mounts
 // based on the object storage type. Currently supported amendments:
+// - All: Ensure object storage secret mounted and auth projected as env vars.
 // - GCS: Ensure env var GOOGLE_APPLICATION_CREDENTIALS in container
 // - S3: Ensure mounting custom CA configmap if any TLSConfig given
 func ConfigureStatefulSet(d *appsv1.StatefulSet, opts Options) error {
 	switch opts.SharedStore {
-	case lokiv1.ObjectStorageSecretGCS:
-		return configureStatefulSet(d, opts.SecretName)
+	case lokiv1.ObjectStorageSecretAlibabaCloud, lokiv1.ObjectStorageSecretAzure, lokiv1.ObjectStorageSecretGCS, lokiv1.ObjectStorageSecretSwift:
+		return configureStatefulSet(d, opts.SecretName, opts.SharedStore)
 	case lokiv1.ObjectStorageSecretS3:
-		if opts.TLS == nil {
-			return nil
+		if err := configureStatefulSet(d, opts.SecretName, opts.SharedStore); err != nil {
+			return err
 		}
 		return configureStatefulSetCA(d, opts.TLS)
 	default:
 		return nil
 	}
 }
 
-// ConfigureDeployment merges a GCS Object Storage volume into the deployment spec.
-// With this, the deployment will expose an environment variable for Google authentication.
-func configureDeployment(d *appsv1.Deployment, secretName string) error {
-	p := ensureCredentialsForGCS(&d.Spec.Template.Spec, secretName)
+// ConfigureDeployment merges the object storage secret volume into the deployment spec.
+// With this, the deployment will expose credentials specific environment variables.
+func configureDeployment(d *appsv1.Deployment, secretName string, t lokiv1.ObjectStorageSecretType) error {
+	p := ensureObjectStoreCredentials(&d.Spec.Template.Spec, secretName, t)
```

Comment:
Use a longer parameter name?

```suggestion
func configureDeployment(d *appsv1.Deployment, secretName string, storeType lokiv1.ObjectStorageSecretType) error {
	p := ensureObjectStoreCredentials(&d.Spec.Template.Spec, secretName, storeType)
```